src/main/webapp/profile.js

```diff
@@ -125,89 +142,188 @@ function editProfile(user, role, document) {
+            otherTopics = user.tutor.skills.slice(otherTopicsIndex + 1);
         }
-        document.getElementById('other-subject').value = otherTopics.join(', ');
+        document.getElementById('other-subject-tutor').value = otherTopics.join(', ');
     } else {
         document.getElementById('tutor-topics').style.display = 'block';
         // Automatically check boxes of topics user has already selected
         if (user.skills.indexOf('math') > -1) {
-            document.getElementById('math').checked = true;
+            document.getElementById('math-tutor').checked = true;
         }
         if (user.skills.indexOf('physics') > -1) {
-            document.getElementById('physics').checked = true;
+            document.getElementById('physics-tutor').checked = true;
         }
         if (user.skills.indexOf('chemistry') > -1) {
-            document.getElementById('chemistry').checked = true;
+            document.getElementById('chemistry-tutor').checked = true;
         }
         if (user.skills.indexOf('biology') > -1) {
-            document.getElementById('biology').checked = true;
+            document.getElementById('biology-tutor').checked = true;
         }
         if (user.skills.indexOf('computer science') > -1) {
-            document.getElementById('computer-science').checked = true;
+            document.getElementById('computer-science-tutor').checked = true;
         }
         if (user.skills.indexOf('social studies') > -1) {
-            document.getElementById('social-studies').checked = true;
+            document.getElementById('social-studies-tutor').checked = true;
         }
         if (user.skills.indexOf('english') > -1) {
-            document.getElementById('english').checked = true;
+            document.getElementById('english-tutor').checked = true;
         }
         if (user.skills.indexOf('spanish') > -1) {
-            document.getElementById('spanish').checked = true;
+            document.getElementById('spanish-tutor').checked = true;
         }
         if (user.skills.indexOf('french') > -1) {
-            document.getElementById('french').checked = true;
+            document.getElementById('french-tutor').checked = true;
         }
         if (user.skills.indexOf('chinese') > -1) {
-            document.getElementById('chinese').checked = true;
+            document.getElementById('chinese-tutor').checked = true;
         }
         var otherTopicsIndex = user.skills.indexOf(' ');
         var otherTopics;
         // Check if any other entries exist
         if (otherTopicsIndex > -1 && user.skills.length > otherTopicsIndex + 1) {
             otherTopics = user.skills.slice(otherTopicsIndex + 1);
         }
-        document.getElementById('other-subject').value = otherTopics.join(', ');
+        document.getElementById('other-subject-tutor').value = otherTopics.join(', ');
+    }
+}
+
+function editProfileStudent(user, both, document) {
+    if (both == true) {
```

Comment:
We can do the same thing here. Store user.student or user in a variable called `student` and just write the if statements once.